minio/minio
# Add metrics for custom auth plugin

## Description

The following per-node iam subsystem metrics for the custom auth plugin are added:

```
plugin_authn_service_last_succ_seconds
  HELP When plugin authentication is configured, returns time (in seconds) since the last successful request to the service
  TYPE gauge

plugin_authn_service_last_fail_seconds
  HELP When plugin authentication is configured, returns time (in seconds) since the last failed request to the service
  TYPE gauge

plugin_authn_service_total_requests_minute
  HELP When plugin authentication is configured, returns total requests count in the last full minute
  TYPE gauge

plugin_authn_service_failed_requests_minute
  HELP When plugin authentication is configured, returns failed requests count in the last full minute
  TYPE gauge

plugin_authn_service_succ_avg_rtt_ms_minute
  HELP When plugin authentication is configured, returns average round-trip-time of successful requests in the last full minute
  TYPE gauge

plugin_authn_service_succ_max_rtt_ms_minute
  HELP When plugin authentication is configured, returns maximum round-trip-time of successful requests in the last full minute
  TYPE gauge
```

## Motivation and Context

Customer requested feature. Provides plugin connectivity and request RTT stats.

## How to test this PR?

The toy authn plugin program in `docs/iam/identity-manager-plugin.go` can be setup for testing this. 
Once setup, make some requests using the custom STS token program in `docs/sts/custom-token-identity.go` to request STS creds (use the token value `aaa` or `bbb` with this program).
New statistics should appear in prometheus.
Can be tested without prometheus setup using `mc admin prometheus metrics myminio`

## Types of changes

## Checklist:
